doc/TAM/Timeslice-Telemetry-and-Granular-Counter-Subscription.md

```diff
@@ -0,0 +1,138 @@
+#  Timeslice Telemetry and Granular Counter Subscription
+-------------------------------------------------------------------------------
+ Title       | Timeslice Telemetry and Granular Counter Subscription
+-------------|-----------------------------------------------------------------
+ Authors     | Jason Bos, Cisco
+ Status      | In review
+ Type        | Standards track
+ Created     | 2023-01-11 - Initial Draft
+ SAI-Version | 1.12
+-------------------------------------------------------------------------------
+
+
+## 1.0  Introduction
+
+This spec enhances the existing TAM (Telemetry and Monitoring) spec to address two new features: timeslice telemetry report format, and granular counter subscription.
+
+The intent is that these may be used together to stream reports with a single counter polled at short intervals, but the API design is such that they could be used
+independently.
+
+## 2.0 Timeslice Telemetry
+
+The Timeslice telemetry format is an output report format for exporting one or more counters correlated with collection timestamps. The goal is to provide a compact format to observe multiple counters correlated closely correlated in time.
+
+In implementations, the reports may be generated by the datapath, so the exact format of the packet will be vendor-defined. One sample format:
+
+| Node ID | Time-0 timestamp | Time-1 timestamp | Time-2 timestamp | Time-3 timestamp | Time-4 timestamp | Time-5 timestamp |
+| :--- | :--- | :--- | :--- | :--- | :--- | :--- |
+| Counter-ID 0 | Counter-0 at Time-0 | Counter-0 at Time-1 | Counter-0 at Time-2 | Counter-0 at Time-3 | Counter-0 at Time-4 | Counter-0 at Time-5  |
+| Counter-ID 1 | Counter-1 at Time-0 | Counter-1 at Time-1 | Counter-1 at Time-2 | Counter-1 at Time-3 | Counter-1 at Time-4 | Counter-1 at Time-5  |
+| Counter-ID 2 | Counter-2 at Time-0 | Counter-2 at Time-1 | Counter-2 at Time-2 | Counter-2 at Time-3 | Counter-2 at Time-4 | Counter-2 at Time-5  |
+
+To configure the format of the report, the TAM report API is extended with a new type and attribute to declare the number of samples per report, i.e. the width of the table.
+
```

Comment:
Can you please give some example for a reference implementation. Something like:


| Switch-id 0x1023456 | UTC 1673848590 | UTC 1673848595 | UTC 1673848600 |
| :--- | :--- | :--- | :--- |
| Eth0-Q2 SAI\_QUEUE\_STAT\_CURR\_OCCUPANCY\_BYTES | 0 | 3000 | 4000 |
| Eth1-Q3 SAI\_QUEUE\_STAT\_CURR\_OCCUPANCY\_BYTES | 4000 | 3000 | 0 |
| Eth2-IPG2 SAI\_INGRESS\_PRIORITY\_GROUP\_STAT\_CURR\_OCCUPANCY\_BYTES | 1000 | 1000 | 1000|